cal/cal/static/js/app.js

```diff
@@ -108,23 +108,100 @@ analyticsApp.component('tags', {
   bindings: {}
 });
 
+function TagsDetailCtrl($scope, $http) {
+  var tagUrl = '/v1/tags/' + this.tagId + '/events';
+  var tagWeek = '/v1/tags/' + this.tagId + '/eventWeek';
+  $scope.tagDetails = [];
+  $scope.tagEvents = [];
+
+  $http({method: 'GET', url: tagUrl + '.json' }).
+  success(function successCallback(data) {
+    for (var i = 0; i < data.results.length; i++) {
+      var event = data.results[i]
+      $scope.tagEvents.push({
+        start: event.start,
+        name: event.name,
+      });
+    }
+  });
+
+  $http({method: 'GET', url: tagWeek + '.json' }).
+  success(function successCallback(data) {
+    var events = [];
+    for (var i = 0; i < data.length; i++) {
+      var event = data[i];
+      var start = new Date(event[0]);
+      events.push({
+        x: start,
+        y: event[1]
+      });
+    }
+    $scope.tagDetails.push({
+      values: events,      //values - represents the array of {x,y} data points
+      key: 'Tag Graph', //key  - the name of the series.
+      color: '#003057',  //color - optional: choose your own line color.
+      strokeWidth: 2,
+    })
+  });
+
+  // line graph
+  $scope.tagLine = {
+    chart: {
+      type: 'lineChart',
+      height: 450,
+      margin : {
+        top: 20,
+        right: 20,
+        bottom: 40,
+        left: 55
+      },
+      x: function(d){ return d.x; },
+      y: function(d){ return d.y; },
+      useInteractiveGuideline: true,
+      xScale: d3.time.scale(),
+      xAxis: {
+        axisLabel: 'Date',
+        tickFormat: function(d) {
+                        return d3.time.format('%m/%d/%y')(d)
+                    }
+      },
+      yAxis: {
+        axisLabel: 'Hours',
+        tickFormat: function(d){
+          return d3.format('.02f')(d);
+        },
+        axisLabelDistance: -10
+      },
+    },
+  };
+};
+
+analyticsApp.component('tagDetail', {
+  templateUrl: '/static/templates/tagDetails.html',
```

Comment:
are we doing `tagDetails` or `tagDetail`? Name of component and name of template should be consistent